invenio_files_rest/helpers.py

```diff
@@ -291,3 +292,15 @@ def create_file(key, path):
                 assert filename.startswith(source)
                 parts = [p for p in filename[len(source):].split(os.sep) if p]
                 yield create_file('/'.join(parts), os.path.join(root, name))
+
+
+def create_file_streaming_redirect_response(obj):
+    """Redirect response generating function."""
+    if obj.deleted:
+        return abort(404)
```

Comment:
I am sorry to come back on this @Glignos :(

I think that right now when a file is deleted the answer is not a 404, but a 200 with `delete_marker: true`.
See here: https://invenio-files-rest.readthedocs.io/en/latest/usage.html#uploading-files

I am not 100%, are you able to test it?
Maybe create a test for this new `if`?

Reply:
So, in this case, the normal behavior would be that the [get_object](https://github.com/inveniosoftware/invenio-files-rest/blob/a7ece14c644a728597db77075feb15e65f88c5e2/invenio_files_rest/views.py#L547) would abort and return a `404` since the [ObjectVersion.get](https://github.com/inveniosoftware/invenio-files-rest/blob/a7ece14c644a728597db77075feb15e65f88c5e2/invenio_files_rest/models.py#L1172) will return `None` since the filters will reject the deleted marker and the previous versions.
Although this still can be removed since it is redundant, since if the object is deleted the request will be aborted beforehand when it is fetched.
Added also the tests since indeed they were missing.